### PR TITLE
Feature/ma 73

### PR DESCRIPTION
**Pull request by dephillipsmichael**
This PR implements the Tapping Task from the ORKOrderedTask class on iOS.

### CODE REVIEW

File: backbone/src/main/java/org/researchstack/backbone/result/TappingIntervalResult.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    public enum TappingButtonIdentifier {
+        // The touch landed outside of the two buttons.
+        TappingButtonIdentifierNone,
+        // The touch landed in the left button.
+        TappingButtonIdentifierLeft,
+        // The touch landed in the right button.
+        TappingButtonIdentifierRight;
+    }
+
+    public static final class Size implements Serializable {
```
Comment (rianhouston) [inline backbone/src/main/java/org/researchstack/backbone/result/TappingIntervalResult.java]: Why did you need to duplicate these classes?  To make them serializable?

File: backbone/src/main/java/org/researchstack/backbone/task/factory/TappingTaskFactory.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                break;
+            case RIGHT:
+                rightHand = true;
+                break;
+            case BOTH:
+                rightHand = !leftFirstIfDoingBoth;
+                break;
+        }
+
+        // Obtain sensor frequency for Walking Task recorders
```
Comment (rianhouston) [inline backbone/src/main/java/org/researchstack/backbone/task/factory/TappingTaskFactory.java]: Walking?  Copy paste?

File: backbone/src/main/java/org/researchstack/backbone/ui/step/layout/TappingIntervalStepLayout.java [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+                        // Make sure we have the same finger's up tap
+                        if (buttonSamples[idx] != null && correctFingerForIdx) {
+                            buttonSamples[idx].setDuration(motionEvent.getDownTime());
+                            sampleList.add(buttonSamples[idx]);
+                            buttonSamples[idx] = null;
+                            lastPointerIdx[idx] = INVALID_POINTER_IDX;
+                            if (countsAsATap) {
+                                countATap();
+                            }
+                            Log.d("TODO_REMOVE", "tap up with idx " + idx);
```
Comment (rianhouston) [inline backbone/src/main/java/org/researchstack/backbone/ui/step/layout/TappingIntervalStepLayout.java]: Use LogExt?

File: backbone/src/main/java/org/researchstack/backbone/ui/views/FixedSubmitBarLayout.java [hunk 4]
```diff
@@ -12,12 +12,15 @@
 
 import org.researchstack.backbone.R;
 import org.researchstack.backbone.ui.step.layout.StepLayout;
+import org.researchstack.backbone.ui.step.layout.TappingIntervalStepLayout;
```
Comment (rianhouston) [inline backbone/src/main/java/org/researchstack/backbone/ui/views/FixedSubmitBarLayout.java]: This isn't needed is it?

**Comment by rianhouston** [comment]
Looks good. :shipit:

File: backbone/src/main/java/org/researchstack/backbone/result/TappingIntervalResult.java (hunk 1 above)
Comment (dephillipsmichael) [reply backbone/src/main/java/org/researchstack/backbone/result/TappingIntervalResult.java]: Yeah, they aren't serializable.  You also can't override them and make them serializable because they are final classes.

Comment (dephillipsmichael) [reply backbone/src/main/java/org/researchstack/backbone/result/TappingIntervalResult.java]: Maybe some comments for them would be a good idea

File: backbone/src/main/java/org/researchstack/backbone/ui/step/layout/TappingIntervalStepLayout.java (hunk 3 above)
Comment (dephillipsmichael) [reply backbone/src/main/java/org/researchstack/backbone/ui/step/layout/TappingIntervalStepLayout.java]: Whoops, will take this out.

File: backbone/src/main/java/org/researchstack/backbone/ui/views/FixedSubmitBarLayout.java (hunk 4 above)
Comment (dephillipsmichael) [reply backbone/src/main/java/org/researchstack/backbone/ui/views/FixedSubmitBarLayout.java]: Nope, will remove
